pkg/monitoring/mimir/service.go

```diff
@@ -29,166 +31,207 @@ type MimirService struct {
+		if err != nil {
+			return "", fmt.Errorf("failed to generate htpasswd for cluster %s: %w", clusterName, err)
+		}
+		htpasswdLines = append(htpasswdLines, htpasswdEntry)
+	}
 
-	return nil
+	return strings.Join(htpasswdLines, "\n"), nil
 }
 
-func (ms *MimirService) CreateApiKey(ctx context.Context, logger logr.Logger) error {
-	objectKey := client.ObjectKey{
-		Name:      mimirApiKey,
-		Namespace: mimirNamespace,
+// createOrUpdateHtpasswdSecret creates or updates an htpasswd secret with the given name and data key
+func (ms *MimirService) createOrUpdateHtpasswdSecret(ctx context.Context, secretName, dataKey string, logger logr.Logger) error {
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: mimirNamespace,
+		},
 	}
 
-	current := &corev1.Secret{}
-	err := ms.Client.Get(ctx, objectKey, current)
-	if apierrors.IsNotFound(err) {
-		// First all secrets using the password from the mimirApiKey secret are deleted
-		// to ensure that they won't use an outdated password.
-		logger.Info("Deleting old secrets")
-
-		err := secret.DeleteSecret(ingressAuthSecretName, mimirNamespace, ctx, ms.Client)
+	result, err := ctrl.CreateOrUpdate(ctx, ms.Client, secret, func() error {
+		// Generate htpasswd content
+		htpasswdContent, err := ms.generateHtpasswdContent(ctx)
 		if err != nil {
-			return fmt.Errorf("failed to delete secret %s/%s: %w", mimirNamespace, ingressAuthSecretName, err)
+			return err
 		}
 
-		// Once all secrets are deleted,the mimirApiKey one may be created.
-		logger.Info("Building auth secret")
-
-		password, err := ms.PasswordManager.GeneratePassword(32)
-		if err != nil {
-			return fmt.Errorf("failed to generate password: %w", err)
+		// Initialize or update the secret data
+		if secret.Data == nil {
+			secret.Data = make(map[string][]byte)
 		}
+		secret.Data[dataKey] = []byte(htpasswdContent)
 
-		secret := secret.GenerateGenericSecret(
-			mimirApiKey, mimirNamespace, "credentials", password)
+		return nil
+	})
 
-		err = ms.Client.Create(ctx, secret)
-		if err != nil {
-			return fmt.Errorf("failed to create secret %s/%s: %w", mimirNamespace, mimirApiKey, err)
-		}
+	if err != nil {
+		return fmt.Errorf("failed to create or update %s secret: %w", secretName, err)
+	}
 
-		logger.Info("Auth secret successfully created")
+	logger.Info("Htpasswd secret processed", "secret", secretName, "result", result)
+	return nil
+}
 
-		return nil
-	} else if err != nil {
-		return fmt.Errorf("failed to get secret %s/%s: %w", mimirNamespace, mimirApiKey, err)
+func (ms *MimirService) configureIngressAuthenticationSecret(ctx context.Context, logger logr.Logger) error {
+	return ms.createOrUpdateHtpasswdSecret(ctx, ingressAuthSecretName, "auth", logger)
+}
+
+func (ms *MimirService) configureHTTPRouteAuthenticationSecret(ctx context.Context, logger logr.Logger) error {
+	return ms.createOrUpdateHtpasswdSecret(ctx, httprouteAuthSecretName, ".htpasswd", logger)
+}
```

Comment:
I don't think we need those, it's another level of abstraction we don't need.

Also can we replace `auth` key with `htpasswd` ? since that's holding htpasswd values.